### PR TITLE
Add J# slanguage

### PR DESCRIPTION
**Pull request by KrzysztofSzewczyk**
## Description

This PR is adding J# language support as stated [there](https://github.com/github/linguist/issues/4124).
It's my "second try", because things gone complicated on the first try.

## Checklist:
- [X] **I am adding a new language.**
  - [X] The extension of the new language is used in hundreds of repositories on GitHub.com.
    (described more in linked issue)
  - [X] I have included a syntax highlighting grammar: Not necessary, using Java one.
  - [X] I have updated the heuristics to distinguish my language from others using the same extension: Other languages with .jsl extension are not warranted support by Linguist, therefore I don't mind them.

## My TODO list:
 - [X] Resolve color conflicts
 - [X] Move samples to valid directory

Also, I ran `update-ids` on languages.yml, though I don't think this is expected result.

A few repositories:
 * https://github.com/matt77hias/jsl-smallpt
 * https://github.com/ankurdave/htw-3d
 * https://github.com/hikerlive/code4me/
 * https://github.com/duchien85/cellengine/blob/4bafd127f579d3cdc1f652500111ad8b88588b8e/gameedit/CellMusicEdit/AppGameBMS/Program.jsl
 * Many, many, many more: https://github.com/search?l=&q=extension%3Ajsl+Program&type=Code
 * Even more: https://github.com/search?l=&q=extension%3Ajsl+class&type=Code

### CODE REVIEW

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@
 # language_id           - Integer used as a language-name-independent indexed field so that we can rename
 #                         languages in Linguist without reindexing all the code on GitHub. Must not be
 #                         changed for existing languages without the explicit permission of GitHub staff.
-# color                 - CSS hex color to represent the language. Only used if type is "programming" or "markup".
+# color                 - CSS hex color to represent the language. Only used if type is "programming" or "prose".
```
Comment (pchaigno) [inline lib/linguist/languages.yml]: Looks like this is reverting #4247 by mistake.

Comment (Alhadis) [reply lib/linguist/languages.yml]: @KrzysztofSzewczyk Could you please do a fresh pull on your branch, making sure that all recent changes this week are in your branch?

Comment (KrzysztofSzewczyk) [reply lib/linguist/languages.yml]: @Alhadis how do I make a fresh pull?

Comment (pchaigno) [reply lib/linguist/languages.yml]: The [SO answer](https://stackoverflow.com/a/3877000/6884590) I've linked to before should work fine for that.

**Comment by pchaigno** [comment]
Flagging as stale.

**Pull request closed without merging**